agent-inject/agent/annotations.go

```diff
@@ -101,6 +101,16 @@ const (
 	// AnnotationAgentRunAsGroup sets the Group ID to run the Vault Agent containers as.
 	AnnotationAgentRunAsGroup = "vault.hashicorp.com/agent-run-as-group"
 
+	// AnnotationAgentRunAsSameUser sets the User ID of the injected Vault Agent
+	// containers to the User ID of the first application container in the Pod.
+	// Requires Spec.Containers[0].SecurityContext.RunAsUser to be set in the
+	// Pod Spec.
+	AnnotationAgentRunAsSameUser = "vault.hashicorp.com/agent-run-as-same-user"
```

Comment:
We're not guaranteed that `Container[0]` is the app container you want to mirror the user of.  I'm thinking pods that have a more than one sidecar container predefined.  This could be a documentation note, but worth thinking about.

Reply:
Yeah, that's a good point. Perhaps this option could specify the name of the container to mirror, instead of just grabbing `Container[0]`. Though that makes it hard to have as a command-line option. 

Maybe a future annotation for specifying the name of the container to grab, if there's demand?

Reply:
Yeah let's just wait to see if folks want it.  I don't think it's unreasonable to suggest putting the container they want first in the spec for now.